### PR TITLE
Picasso builder

### PR DESCRIPTION
**Pull request by jaredsburrows**
Closes https://github.com/square/picasso/issues/1750
@JakeWharton @NightlyNexus

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 1]
```diff
@@ -664,13 +668,24 @@ void shutdown() {
     private ExecutorService service;
     private Cache cache;
     private Listener listener;
-    private RequestTransformer transformer;
-    private List<RequestHandler> requestHandlers;
+    private RequestTransformer requestTransformer;
+    private List<RequestHandler> requestHandlers = new ArrayList<>();
```
Comment (jaredsburrows) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: @JakeWharton should I expose `requestHandlers` via a getter so verify it's contents in the `newBuilder` test?

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 2]
```diff
@@ -664,13 +668,24 @@ void shutdown() {
...
+    Builder(@NonNull Picasso picasso) {
+      context = picasso.context;
+      cache = picasso.cache;
+      listener = picasso.listener;
+      requestTransformer = picasso.requestTransformer;
+      requestHandlers.addAll(picasso.requestHandlers);
+      defaultBitmapConfig = picasso.defaultBitmapConfig;
+      indicatorsEnabled = picasso.indicatorsEnabled;
+      loggingEnabled = picasso.loggingEnabled;
+    }
```
Comment (NightlyNexus) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: i need the OkHttpClient to be shared.

Comment (jaredsburrows) [reply picasso/src/main/java/com/squareup/picasso/Picasso.java]: I will look into exposing `OkHttpClient` since it initiated in `OkHttp3Downloader `.

Comment (jaredsburrows) [reply picasso/src/main/java/com/squareup/picasso/Picasso.java]: Maybe we can get this in as a first step? I see you created another issue to remove `Downloader`.

**Comment by jrodbx** [comment]
Closing, since https://github.com/square/picasso/issues/1750 is closed.

**Pull request closed without merging**